Backend/Backend.WebServices/Controllers/EnterpriseController.cs

```diff
@@ -53,24 +54,31 @@ public IEnumerable<EnterpriseDTO> Get()
         /// <param name="id">ID of the enterprise whose details will be returned</param>
         /// <param name="password">Password of the enterprise</param>
         /// <returns>An EnterpriseDTO with the specified ID</returns>
-        public EnterpriseDTO Get(int id, string password = null)
+        public EnterpriseDetailsDTO Get(int id)
         {
             ThrowExceptionIfNotHttps();
-            var enterprises = _context.Enterprises.ToList();
-            Enterprise enterprise = null;
+            var enterprise = _context.Enterprises
+                                .Where(x => x.Id == id)
+                                .Include(x => x.Participants.Select(
+                                         p => p.Places.Select(
+                                         m => m.MediaItems.Select(
+                                         t => t.MediaItemType))))
+                                .SingleOrDefault();
 
-            foreach (Enterprise e in enterprises)
-                if (e.Id == id)
-                    enterprise = e;
+            if (enterprise == null)
+                throw new HttpException(404, "Enterprise not found. The Enterprise with ID: " + id + " does not exist."); //Might be better to do something else
 
-            if (password != null && enterprise.Password != null)
-                if (!enterprise.Password.Equals(password))
-                    throw new HttpException(403, "Password is incorrect");
+            string password = null;
 
-            if (enterprise == null)
-                throw new HttpException(404, "Enterprise not found");
+            if (Request.Headers.Any(x => "Authorization".Equals(x.Key)))
+            {
+                password = Request.Headers.GetValues("Authorization").First();
+
+                if (!password.Equals(enterprise.Password))
```

Comment:
Do we care if the client sends a password for an enterprise that doesn't have one? What if they sent an Authorization header with an empty value in cases where the enterprise doesn't have a password? (Then again, maybe I'm just being nitpicky :/ )